### PR TITLE
Use session more and specify error codes

### LINKED ISSUE #238: Ensure use of connection/session are clear and specify error codes when closing a session.
Part of PR #234 has text "MUST close the connection if there are duplicates"

It seems like we should be closing the session, and we should specify an error code.

### PR DESCRIPTION
**Pull request by ianswett**
Fixes #238 

I also wonder if Connection URL should be renamed to Session URL?

### CODE REVIEW

File: draft-ietf-moq-transport.md [hunk 1]
```diff
@@ -810,7 +810,7 @@ The client offers the list of the protocol versions it supports; the
 server MUST reply with one of the versions offered by the client. If the
 server does not support any of the versions offered by the client, or
 the client receives a server version that it did not offer, the
-corresponding peer MUST close the connection.
+corresponding peer MUST close the session.
```
Comment (ianswett) [inline draft-ietf-moq-transport.md:813]: Should we add a 'VN Failed' error code?

Comment (kixelated) [reply draft-ietf-moq-transport.md:813]: I think so.

**Review by kixelated — APPROVED** [review]
I would prefer:

> close the session with a 'Protocol Violation'

instead of:

> close the session as a 'Protocol Violation'

**Review by fluffy — APPROVED** [review]
LGTM

File: draft-ietf-moq-transport.md (hunk 1 above)
Comment (ianswett) [reply draft-ietf-moq-transport.md:813]: SG, will do in a different PR.

**Pull request merged**